spring-beans/src/main/java/org/springframework/beans/factory/support/AbstractAutowireCapableBeanFactory.java

```diff
@@ -598,12 +598,12 @@ protected Object doCreateBean(String beanName, RootBeanDefinition mbd, @Nullable
 			populateBean(beanName, mbd, instanceWrapper);
 			exposedObject = initializeBean(beanName, exposedObject, mbd);
 		}
-		catch (Throwable ex) {
-			if (ex instanceof BeanCreationException && beanName.equals(((BeanCreationException) ex).getBeanName())) {
-				throw (BeanCreationException) ex;
+		catch (Throwable th) {
```

Comment:
Please do not make stylistic change like that. We name our exception `ex` regardless of the type.

Reply:
Following up on what @snicoll said, we'd likely rewrite that similar to the following:

```java
if (ex instanceof BeanCreationException bce && beanName.equals(bce.getBeanName())) {
	throw bce;
}
```

Reply:
@snicoll @sbrannen  done